griptape/tasks/prompt_task.py

```diff
@@ -68,8 +68,9 @@ def prompt_driver(self) -> BasePromptDriver:
 
     def preprocess(self, structure: Structure) -> PromptTask:
         super().preprocess(structure)
+
         if self.prompt_driver is not None:
-            self.prompt_driver.structure = structure
+            self.prompt_driver.add_event_listeners(self.structure.event_listeners)
```

Comment:
Calling this out. Is this weird? Should all tasks do this?

Reply:
To me, it doesn't actually feel that much weirder than what was previously there, but the new way will result in different behavior if say you modify the event_listeners on the structure after the task was added to the structure. That wouldn't be that intuitive:

```py
agent = Agent(tasks=[PromptTask(...)])
agent.add_event_listener(...)
# now prompt_task.prompt_driver doesn't have the same event_listeners as the agent
```

Reply:
Yeah that's actually why I had to do [this](https://github.com/griptape-ai/griptape/blob/3622d31f20121e60b4e7878ef3f2aceedf401b71/tests/unit/events/test_event_listener.py#L42-L43) which I don't love. 

Any suggestions?

Reply:
Let me think about this for a couple minutes. I feel like this is going to make it way easier to shoot ourselves in the feet.

Reply:
Here's an idea: What if the structure adds an event listener to the drivers that just forwards the events to its own event listeners? The added event listener would have a reference to the structure, so nothing will ever get out of sync. If you add an event listener to the structure, it would still just work

Reply:
Check out the latest implementation, it cleaned up a few areas.